### PR TITLE
rgw/notification: add amqp 1.0 endpoint creation in rgw

### PR DESCRIPTION
**Pull request by wangxuw**
add qpid-proton amqp 1.0 endpoint creation in rgw and allow publishing to amqp 1.0 endpoints

Signed-off-by: Xu Wang (wilyxerus@gmail.com)

### CODE REVIEW

File: src/rgw/rgw_amqp_1.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab ft=cpp
```
Comment (yuvalif) [inline src/rgw/rgw_amqp_1.h:2]: nit: please use 2 spaces instead of tabs for indentation

**Comment by yuvalif** [comment]
please add signoff top the commit message

**Comment by yuvalif** [comment]
why do we need the list of connections (with the mutex)?
if the connections are handled internally in the library, maybe we don't need to manage that in our code as well?
if we do need to keep this list, we should be able to remove from it when a connections is closed

**Review by yuvalif — COMMENTED** [review]
please see comments in the code

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by mattbenjamin** [comment]
@yuvalif I can't remember what we said about this;  we have something, but didn't find a way to get it merged?

**Comment by yuvalif** [comment]
> @yuvalif I can't remember what we said about this; we have something, but didn't find a way to get it merged?

code is done and there is also unit testing in place.
however, more tsting is needed, and there are crashes in some cases.
also, documentation is missing. and teuthology tests are missing.

**Pull request closed without merging**